### PR TITLE
Add pull request template

### PR DESCRIPTION
**Pull request by delucis**
This PR suggests adding a template that will be used to pre-fill the PR description when someone opens a new one. It includes some advice and information based on various things that have come up in discussion recently.

What else would it be useful to mention here? Could anything be clearer? And does anything feel superfluous so we can slim this down?

### CODE REVIEW

**Comment by sarah11918** [comment]
I might not ask "Is this a minor change?" but rather something like, "Does this change require a change in all other languages, too?" (If not, be sure to include a keyword like "minor" or "typo"....)  ??

**Comment by sarah11918** [comment]
I might selfishly also like a prompt to TITLE the PR in a certain way, if it's an i18n translation? The way most of the languages have already been doing it.

I haven't yet thought about what to do about /en/ text PRs, but it's only fair we should probably devise something consistent, too.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<!-- Thank you for opening a PR! We really appreciate you taking the time to help out 🙌 -->
+
+#### Changes
+
+- What does this PR change? Give us a brief description.
+- Did you change something visual? A before/after screenshot can be helpful.
+- Is this a minor change like fixing a link or a typo?
```
Comment (delucis) [inline .github/PULL_REQUEST_TEMPLATE.md:7]:
```suggestion
- Is this change small enough not impact translation work?
```

**Comment by hippotastic** [comment]
For i18n PRs, you could add a suggested PR title format. It seems that many PRs are now using something like this, and I love those standardized titles:
```md
i18n(xy): Add `path/to/page-name.md` translation
     ^^-- language code, e.g. "de", "es", "pt-BR"
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<!-- Thank you for opening a PR! We really appreciate you taking the time to help out 🙌 -->
+
+#### Changes
+
+- What does this PR change? Give us a brief description.
+- Did you change something visual? A before/after screenshot can be helpful.
+- Is this change small enough not impact translation work?
+  Include `minor`, `broken link` or `typo` in the PR title
+- Is this a translation PR? Consider pinging your i18n-gang on Discord for a review.
```
Comment (delucis) [inline .github/PULL_REQUEST_TEMPLATE.md:9]:
```suggestion
- Is this a translation PR?  
  - It’s helpful to title your PR something like “i18n(lang): PR description”
  - Consider pinging your i18n-gang on Discord for a review.
```

**Comment by zadeviggers** [comment]
I suggest adding a checklist with common types of changes and giving them the option to write more if it's a complex or unsual change, rather than requiring lots of writing for small changes.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (delucis) [reply .github/PULL_REQUEST_TEMPLATE.md:9]: @hippotastic @sarah11918 How’s this look?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+<!-- Thank you for opening a PR! We really appreciate you taking the time to help out 🙌 -->
+
+#### Changes
+
+- What does this PR change? Give us a brief description.
+- Did you change something visual? A before/after screenshot can be helpful.
+- Is this change small enough not impact translation work?
```
Comment (hippotastic) [inline .github/PULL_REQUEST_TEMPLATE.md:7]:
```suggestion
- Is this change small enough not to impact translation work?
```

**Comment by delucis** [comment]
> I suggest adding a checklist with common types of changes and giving them the option to write more if it's a complex or unsual change, rather than requiring lots of writing for small changes.

Love this idea. Here’s my first pass. Any other categories come to mind?

```
#### What kind of changes does this PR include?

- [ ] Minor content fixes (broken links, typos, etc.)
- [ ] New or updated content
- [ ] Translated content
- [ ] Changes to the docs site code
- [ ] Something else!
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  Include `minor`, `broken link` or `typo` in the PR title
```
Comment (hippotastic) [inline .github/PULL_REQUEST_TEMPLATE.md:8]:
```suggestion
  If so, please include `minor`, `broken link` or `typo` in the PR title
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+- Not on Discord? Join us at https://astro.build/chat
+
+<!--
+Here’s what will happen next:
+
+1. Our GitHub bots will run to check your changes.
+   If they spot any broken links you will see some error messages on this PR.
+   Don’t hesitate to ask any questions if you’re not sure what these mean!
+
+2. You’ll be able to see a preview of you changes on Netlify 🥳
```
Comment (hippotastic) [inline .github/PULL_REQUEST_TEMPLATE.md:19]:
```suggestion
2. In a few minutes, you’ll be able to see a preview of your changes on Netlify 🥳
```

**Review by hippotastic — COMMENTED** [review]
Looks good overall! Just a few minor suggestions :)

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
+<!-- Thank you for opening a PR! We really appreciate you taking the time to help out 🙌 -->
+
+#### Changes
```
Comment (delucis) [inline .github/PULL_REQUEST_TEMPLATE.md:3]:
```suggestion
#### What kind of changes does this PR include?

<!-- Place an X in the [ ] for any of these that apply -->

- [ ] Minor content fixes (broken links, typos, etc.)
- [ ] New or updated content
- [ ] Translated content
- [ ] Changes to the docs site code
- [ ] Something else!

#### Description
```

**Comment by hippotastic** [comment]
> Love this idea. Here’s my first pass. Any other categories come to mind?

I also love this idea. It makes it simpler for contributors, which is great by itself, and it might even allow us to use this checklist in the automation to decide whether an update to an English source page is "major" or not. Especially the scenario of one PR including both typo fixes AND major content changes could be solved by a "tick all that apply" checklist and would be a big advantage to matching magic keywords in the commit titles.

**Comment by sarah11918** [comment]
Great, great ideas everyone!

My only thought right now is that maybe I could envision a case where a "minor" change actually DOES require every language to update?

I'm still +1 on somehow determining that a change should affect only the language of the PR somehow, if there's a good way to do that. BUT, I recognize that Chris and I will have the ability to manually determine whether a title is going to work the way we want it to, and can fix it on a case-by-case basis.  :)

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+#### Description
+
+- What does this PR change? Give us a brief description.
+- Did you change something visual? A before/after screenshot can be helpful.
+- Is this change small enough not to impact translation work?
+  If so, please include `minor`, `broken link` or `typo` in the PR title
+- Is this a translation PR?  
+  - It’s helpful to title your PR something like “i18n(lang): PR description”
+  - Consider pinging your i18n-gang on Discord for a review.
+- Not on Discord? Join us at https://astro.build/chat
```
Comment (delucis) [inline .github/PULL_REQUEST_TEMPLATE.md:21]: Removing this more detailed advice so we don’t overload people. We should include it in a contributing guide when we have one. We can retitle PRs manually if necessary based on the answers to the checklist.

```suggestion
```

**Comment by hippotastic** [comment]
Maybe we should add an explicit item like this to make contributors to English docs think about whether their change is substantial enough to warrant retranslations:
```md
- [ ] Changes to the English documentation that should be updated in translations as well
```

**Comment by delucis** [comment]
> Maybe we should add an explicit item like this to make contributors to English docs think about whether their change is substantial enough to warrant retranslations

So I was thinking that it's easy for someone to answer "I did X" but more complex to answer "Doing X impacts other people like..." Because for the latter you need to have a much better understanding of the whole repo, methodologies, and people involved. So I was trying to ask questions that would be easy to answer and would give maintainers a good idea of the answer to your question.

**Comment by sarah11918** [comment]
As much as I love @hippotastic 's question, I think @delucis is right that we can't assume every English PR'er can make that determination, and that's on us.

I think at least having them indicate typo/broken link kind of stuff gives the maintainers a head start.  :)

**Comment by hippotastic** [comment]
Gotcha! It was just an idea, I'm totally fine with not including it in order not to overload contributors with tons of difficult choices.

**Comment by sarah11918** [comment]
I'm loving the checklist! ❤️

**Comment by sarah11918** [comment]
@delucis Do you need anything else for this? I'm happy if you are!

**Comment by delucis** [comment]
Don't think so — I was waiting for more ideas or an LGTM 👍

**Pull request merged**